From 397586625e71d166f493f16bfe04f3005ae464c3 Mon Sep 17 00:00:00 2001
From: Olivier Brunel <jjk@jjacky.com>
Date: Sat, 13 Aug 2016 20:31:49 +0200
Subject: [PATCH 4/4] getdtablesize: fix returning hard instead of soft rlimit

This makes the result consistent with sysconf(_SC_OPEN_MAX).
---
 src/legacy/getdtablesize.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/legacy/getdtablesize.c b/src/legacy/getdtablesize.c
index 682da6d..b30c193 100644
--- a/src/legacy/getdtablesize.c
+++ b/src/legacy/getdtablesize.c
@@ -7,5 +7,5 @@ int getdtablesize(void)
 {
 	struct rlimit rl;
 	getrlimit(RLIMIT_NOFILE, &rl);
-	return rl.rlim_max < INT_MAX ? rl.rlim_max : INT_MAX;
+	return rl.rlim_cur < INT_MAX ? rl.rlim_cur : INT_MAX;
 }
-- 
2.10.1

